src/devtools/client/webconsole/reducers/messages.ts

```diff
@@ -285,6 +282,26 @@ const messagesSlice = createSlice({
         state as MessageState
       );
     },
+    clearMessages(state) {
+      // Note that we don't use removeMessagesFromState() in this case, because we're removing ALL messages.
+      // That method does extra work to handle the case where we are only removing SOME messages.
+      messagesAdapter.removeAll((state as MessageState).messages);
```

Comment:
:hand: Do we need to reset `state.logpointMessages` here too?

Tbh I'm still not entirely sure what that field is used for, and looking at my local copy it _seems_ like it's only being used within this file.  But it seems like a thing we'd want to clear out in here also.

Reply:
Hmm, I'm actually not sure. I _think_ log points are local (in memory) as opposed to messages that are fetched from the server. Would we need to clear log points out before refetching messages? Maybe @jaril or @jcmorrow knows?

Reply:
I just tried with and without clearing the `logpointMessages` and I think it's kind of a weird user experience either way. The log point either disappears or becomes kind of half broken when you change the focus.

I'm not sure I understand the purpose of this bit of message state. Looks like it's been around since Brian Hackett imported the code ~two years ago in fc08cf873c1844587d97895bf89174eca4bf511c.

Reply:
I think it _is_ referenced in quite a few places though I'm not yet sure what it's used for. This is what removing it would look like, I think:
```diff
diff --git a/src/devtools/client/debugger/src/actions/breakpoints/logpoints.ts b/src/devtools/client/debugger/src/actions/breakpoints/logpoints.ts
index dbe648c4a..4c980328a 100644
--- a/src/devtools/client/debugger/src/actions/breakpoints/logpoints.ts
+++ b/src/devtools/client/debugger/src/actions/breakpoints/logpoints.ts
@@ -15,9 +15,6 @@ import { _removeBreakpoint, removeBreakpointOption } from "./modify";
 export function removeLogpointsInSource(cx: Context, source: Source): UIThunkAction {
   return async (dispatch, getState) => {
     const breakpoints = getLogpointsForSource(getState(), source.id);
-    for (const breakpoint of breakpoints) {
-      dispatch(removeLogpoint(cx, breakpoint));
-    }
     const requestedBreakpointLocations = getRequestedBreakpointLocations(getState());
     for (const location of Object.values(requestedBreakpointLocations)) {
       if (location.sourceId === source.id) {
@@ -29,11 +26,6 @@ export function removeLogpointsInSource(cx: Context, source: Source): UIThunkAct
 
 export function toggleLogpoint(cx: Context, line: number, bp?: Breakpoint): UIThunkAction {
   return dispatch => {
-    if (bp?.options.logValue) {
-      trackEvent("breakpoint.minus_click");
-      return dispatch(removeLogpoint(cx, bp));
-    }
-
     trackEvent("breakpoint.plus_click");
     return dispatch(addLogpoint(cx, line));
   };
diff --git a/src/devtools/client/debugger/src/client/commands.ts b/src/devtools/client/debugger/src/client/commands.ts
index ac391b5b1..9aff7bc89 100644
--- a/src/devtools/client/debugger/src/client/commands.ts
+++ b/src/devtools/client/debugger/src/client/commands.ts
@@ -15,7 +15,6 @@ import {
   newLogGroupId,
   setEventLogpoints,
   setExceptionLogpoint,
-  removeLogpoint,
 } from "protocol/logpoint";
 import { ThreadFront, createPrimitiveValueFront, ValueFront } from "protocol/thread";
 import { WiredNamedValue } from "protocol/thread/pause";
@@ -115,19 +114,11 @@ function maybeGenerateLogGroupId(options: InitialBreakpointOptions): FinalBreakp
   return options;
 }
 
-async function maybeClearLogpoint(location: SourceLocation) {
-  const bp = breakpoints[locationKey(location)];
-  if (bp && bp.options.logGroupId) {
-    removeLogpoint(bp.options.logGroupId);
-  }
-}
-
 function hasBreakpoint(location: SourceLocation) {
   return !!breakpoints[locationKey(location)];
 }
 
 function setBreakpoint(location: SourceLocation, options: InitialBreakpointOptions) {
-  maybeClearLogpoint(location);
   const finalOptions = maybeGenerateLogGroupId(options);
   breakpoints[locationKey(location)] = { location, options: finalOptions };
 
@@ -157,7 +148,6 @@ function setBreakpoint(location: SourceLocation, options: InitialBreakpointOptio
 }
 
 function removeBreakpoint(location: SourceLocation) {
-  maybeClearLogpoint(location);
   delete breakpoints[locationKey(location)];
 
   const { line, column, sourceUrl, sourceId } = location;
@@ -320,9 +310,6 @@ function setEventListenerBreakpoints(ids: string[]) {
 let gExceptionLogpointGroupId: string | null;
 
 function logExceptions(shouldLog: boolean) {
-  if (gExceptionLogpointGroupId) {
-    removeLogpoint(gExceptionLogpointGroupId);
-  }
   if (shouldLog) {
     gExceptionLogpointGroupId = newLogGroupId();
     setExceptionLogpoint(gExceptionLogpointGroupId);
diff --git a/src/devtools/client/debugger/src/components/SecondaryPanes/LogpointsPane.tsx b/src/devtools/client/debugger/src/components/SecondaryPanes/LogpointsPane.tsx
index e139342f4..693e3e860 100644
--- a/src/devtools/client/debugger/src/components/SecondaryPanes/LogpointsPane.tsx
+++ b/src/devtools/client/debugger/src/components/SecondaryPanes/LogpointsPane.tsx
@@ -3,7 +3,7 @@ import { useDispatch, useSelector } from "react-redux";
 import { getLogpointSources } from "../../selectors/breakpointSources";
 import Breakpoints from "./Breakpoints";
 import MaterialIcon from "ui/components/shared/MaterialIcon";
-import { removeLogpoint, removeLogpointsInSource } from "../../actions/breakpoints/logpoints";
+import { removeLogpointsInSource } from "../../actions/breakpoints/logpoints";
 import { Context } from "../../selectors";
 import { Breakpoint, Source } from "../../reducers/types";
 
@@ -25,12 +25,8 @@ export default function LogpointsPane() {
       type="print-statement"
       emptyContent={emptyContent}
       breakpointSources={logpointSources}
-      onRemoveBreakpoint={(cx: Context, breakpoint: Breakpoint) =>
-        dispatch(removeLogpoint(cx, breakpoint))
-      }
-      onRemoveBreakpoints={(cx: Context, source: Source) =>
-        dispatch(removeLogpointsInSource(cx, source))
-      }
+      onRemoveBreakpoint={(cx: Context, breakpoint: Breakpoint) => {}}
+      onRemoveBreakpoints={(cx: Context, source: Source) => {}}
     />
   );
 }
diff --git a/src/devtools/client/webconsole/actions/messages.ts b/src/devtools/client/webconsole/actions/messages.ts
index ae555aa78..c69bb8237 100644
--- a/src/devtools/client/webconsole/actions/messages.ts
+++ b/src/devtools/client/webconsole/actions/messages.ts
@@ -21,7 +21,6 @@ import { isFocusRegionSubset } from "ui/utils/timeline";
 
 import {
   clearMessages,
-  logpointMessagesCleared,
   Message as InternalMessage,
   messagesAdded,
   messageClosed,
@@ -42,7 +41,6 @@ export function setupMessages(store: UIStore, ThreadFront: typeof ThreadFrontTyp
     store.dispatch(onLogpointLoading(logGroupId, point, time, location));
   LogpointHandlers.onResult = (logGroupId, point, time, location, pause, values) =>
     store.dispatch(onLogpointResult(logGroupId, point, time, location, pause, values));
-  LogpointHandlers.clearLogpoint = logGroupId => store.dispatch(messagesClearLogpoint(logGroupId));
   TestMessageHandlers.onTestMessage = msg => store.dispatch(onConsoleMessage(msg));
 
   ThreadFront.findConsoleMessages(
@@ -235,7 +233,6 @@ export function messagesAdd(
 }
 
 export const messagesClearEvaluations = messageEvaluationsCleared;
-export const messagesClearLogpoint = logpointMessagesCleared;
 export const messageOpen = messageOpened;
 export const messageClose = messageClosed;
 
diff --git a/src/devtools/client/webconsole/reducers/messages.ts b/src/devtools/client/webconsole/reducers/messages.ts
index 7689e9b33..c82809c31 100644
--- a/src/devtools/client/webconsole/reducers/messages.ts
+++ b/src/devtools/client/webconsole/reducers/messages.ts
@@ -95,11 +95,6 @@ export interface Message {
   };
 }
 
-interface LogpointMessageEntry {
-  messageId: MessageId;
-  key: string;
-}
-
 interface FilteredMessagesCount {
   debug: number;
   info: number;
@@ -141,8 +136,6 @@ export interface MessageState {
   filteredMessagesCount: FilteredMessagesCount;
   /** List of message IDs that are opened */
   messagesUiById: MessageId[];
-  /** Lookup table of which messages are "logpoint" messages */
-  logpointMessages: EntityState<LogpointMessageEntry>;
   /** Set of logpoint IDs that have been removed */
   removedLogpointIds: MessageId[];
   /** Any execution point we are currently paused at, when replaying. */
@@ -171,9 +164,6 @@ export const appendToHistory = (command: Command, history: Command[]): Command[]
 };
 
 export const messagesAdapter = createEntityAdapter<Message>();
-const logpointMessagesAdapter = createEntityAdapter<LogpointMessageEntry>({
-  selectId: entry => entry.key,
-});
 
 export const syncInitialMessageState = (overrides: Partial<MessageState> = {}): MessageState => {
   // Realistically, we only expect filters and commandHistory
@@ -190,7 +180,6 @@ export const syncInitialMessageState = (overrides: Partial<MessageState> = {}):
         messagesUiById: [],
         commandHistory: [],
         exceptionLogpointError: null,
-        logpointMessages: logpointMessagesAdapter.getInitialState(),
         removedLogpointIds: [],
         pausedExecutionPoint: null,
         pausedExecutionPointTime: 0,
@@ -248,20 +237,6 @@ const messagesSlice = createSlice({
 
       return removeMessagesFromState(state as MessageState, removedIds);
     },
-    logpointMessagesCleared(state, action: PayloadAction<string>) {
-      const logpointId = action.payload;
-      const removedIds = [];
-
-      for (const [id, message] of Object.entries(state.messages.entities)) {
-        if (message!.logpointId == logpointId) {
-          removedIds.push(id);
-        }
-      }
-
-      state.removedLogpointIds = Array.from(new Set([...state.removedLogpointIds, logpointId]));
-
-      return removeMessagesFromState(state as MessageState, removedIds);
-    },
     exceptionLogpointErrorCleared(state) {
       state.exceptionLogpointError = null;
     },
@@ -323,7 +298,6 @@ const messagesSlice = createSlice({
 
 export const {
   clearMessages,
-  logpointMessagesCleared,
   messageClosed,
   messageEvaluationsCleared,
   messageOpened,
@@ -366,22 +340,6 @@ function addMessage(newMessage: Message, state: MessageState): MessageState {
     state.hasExecutionPoints = true;
   }
 
-  // When replaying, we might get two messages with the same execution point and
-  // logpoint ID. In this case the first message is provisional and should be
-  // removed.
-  const removedIds = [];
-  if (newMessage.logpointId || (newMessage.evalId && newMessage.type === MESSAGE_TYPE.RESULT)) {
-    const key = `${newMessage.logpointId || newMessage.evalId}:${newMessage.executionPoint}`;
-    const existingMessage = state.logpointMessages.entities[key];
-    if (existingMessage) {
-      log(`LogpointFinish ${newMessage.executionPoint}`);
-      removedIds.push(existingMessage.messageId);
-    } else {
-      log(`LogpointStart ${newMessage.executionPoint}`);
-    }
-    logpointMessagesAdapter.upsertOne(state.logpointMessages, { key, messageId: newMessage.id });
-  }
-
   // Immer will freeze anyway, but this might help skipping the `ValueFront` fields
   const addedMessage = Object.freeze(newMessage);
   messagesAdapter.upsertOne(state.messages, addedMessage);
@@ -399,7 +357,7 @@ function addMessage(newMessage: Message, state: MessageState): MessageState {
     state.filteredMessagesCount[addedMessage.level as FilterCountKeys]++;
   }
 
-  return removeMessagesFromState(state, removedIds);
+  return state;
 }
 
 interface MessagesFilters {
diff --git a/src/protocol/logpoint.ts b/src/protocol/logpoint.ts
index e0609c1cc..75bca6970 100644
--- a/src/protocol/logpoint.ts
+++ b/src/protocol/logpoint.ts
@@ -38,7 +38,6 @@ export const LogpointHandlers: {
     time: number,
     location: Location
   ) => void;
-  clearLogpoint?: (logGroupId: string) => void;
 } = {};
 
 let store: UIStore;
@@ -433,7 +432,6 @@ export function newLogGroupId() {
 export async function setEventLogpoints(eventTypes: string[]) {
   for (const eventType in eventTypeLogGroupId) {
     if (eventTypes.indexOf(eventType) < 0) {
-      removeLogpoint(eventTypeLogGroupId[eventType]);
       delete eventTypeLogGroupId[eventType];
     }
   }
@@ -533,7 +531,6 @@ export async function setExceptionLogpoint(logGroupId: string) {
   try {
     await analysisManager.runAnalysis(params, handler);
   } catch (e: any) {
-    removeLogpoint(logGroupId);
     let msg;
     if (e.code === ProtocolError.TooManyPoints) {
       msg = "There are too many exceptions. Please focus to a smaller time range and try again.";
@@ -544,9 +541,3 @@ export async function setExceptionLogpoint(logGroupId: string) {
     store.dispatch(exceptionLogpointErrorReceived(msg));
   }
 }
-
-export function removeLogpoint(logGroupId: string) {
-  if (LogpointHandlers.clearLogpoint) {
-    LogpointHandlers.clearLogpoint(logGroupId);
-  }
-}
```

Reply:
@bvaughn can you describe what the "half-broken" state that you're seeing is?

Reply:
What I was seeing earlier (and will dig into in the morning) is that the logpoint UI would stick around, but the messages would get cleared from the console and never show up again _unless_ you removed and re-saved the logpoint. I have some ideas about why this is (though I'm still not sure I understand the purpose of the piece of state we're discussing above).

Reply:
Update: I was doing something stupid (clearing _all_ messages in memory, including log points). Since log point state is stored in a few places, the log point overlay stuck around but the log points never came back in the console.

I've fixed it with the most recent push. I'll probably go ahead and land this in a few minutes, assuming CI is happy, and we can iterate.